framework/src/utils/MooseEnum.C

```diff
@@ -64,6 +65,30 @@ MooseEnum::operator=(const std::string & name)
   return *this;
 }
 
+MooseEnum &
+MooseEnum::operator=(int value)
+{
+  if (value == MooseEnumItem::INVALID_ID)
+  {
+    _current = MooseEnumItem("", MooseEnumItem::INVALID_ID);
+    return *this;
+  }
+
+  std::set<MooseEnumItem>::const_iterator iter = find(value);
+  if (iter == _items.end())
+    mooseError(std::string("Invalid id \""),
```

Comment:
Constructing a std::string is no longer necessary now that we aren't using `std::string::operator+`